src/test/java/io/specto/hoverfly/junit/core/model/DiffViewTest.java

```diff
@@ -0,0 +1,62 @@
+package io.specto.hoverfly.junit.core.model;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.Resources;
+import io.specto.hoverfly.junit.api.view.DiffView;
+import java.net.URL;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DiffViewTest {
+
+    private URL emptyDiffsJson = Resources.getResource("diff/empty-diffs.json");
+    private URL recordedDiffsJson = Resources.getResource("diff/recorded-diffs.json");
+
+    @Test
+    public void shouldDeserializeDiffJsonAndCreateDiffMessage() throws Exception {
+        // Given
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        // When
+        DiffView actual = objectMapper.readValue(recordedDiffsJson, DiffView.class);
+
+        // Then
+        assertThat(actual.getDiffs()).hasSize(2);
+        assertThat(actual.getDiffs().get(0).getDiffReports()).hasSize(2);
```

Comment:
might be better to use some local variable to reduce clutter in these assertions.

Reply:
refactored in 8f3c4f7